### PR TITLE
Release 3.4.3

### PR DESCRIPTION
**Pull request by maxulysse**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/sarek/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/sarek _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nf-test test tests/ --verbose --profile +docker`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by maxulysse** [comment]
Expected snapshots in HaplotypeCaller are dropped in #1596 , as we will be updating all that with nf-test, I'd rather kill the old functionalities than keeping them alive.

**Comment by asp8200** [comment]
I see the `usage.md` has been updated. I had a quick look here https://nf-co.re/sarek/dev/docs/usage  and it seems to render fine.

File: subworkflows/local/annotation_cache_initialisation/main.nf [hunk 1]
```diff
@@ -39,7 +40,8 @@ workflow ANNOTATION_CACHE_INITIALISATION {
 
     if (vep_enabled) {
         def vep_annotation_cache_key = (vep_cache == "s3://annotation-cache/vep_cache/") ? "${vep_cache_version}_${vep_genome}/" : ""
-        def vep_cache_dir = "${vep_annotation_cache_key}${vep_species}/${vep_cache_version}_${vep_genome}"
+        def vep_species_suffix = vep_custom_args.contains("--merged") ? '_merged' : (vep_custom_args.contains("--refseq") ? '_refseq' : '')
+        def vep_cache_dir = "${vep_annotation_cache_key}${vep_species}${vep_species_suffix}/${vep_cache_version}_${vep_genome}"
```
Comment (asp8200) [inline subworkflows/local/annotation_cache_initialisation/main.nf:44]: Has this change been tested? (I don't suppose we have a CI-test for that, right?)

**Comment by asp8200** [comment]
Looks good. Not too many changes. A small release - the way I like it 😁.  (I didn't look at the CNVKIT subworkflow, so hopefully somebody else will take a critical look at that.)

File: subworkflows/local/bam_variant_calling_cnvkit/main.nf [hunk 2]
```diff
@@ -21,12 +23,23 @@ workflow BAM_VARIANT_CALLING_CNVKIT {
...
 
+    // right now we do not use an input VCF to improve the calling of B alleles
+    // based on SNV frequencies from the VCF file
+    // in the future we might consider to add this, by connecting the emission from
+    // SNV variant calling modules
+    CNVKIT_CALL(CNVKIT_BATCH.out.cns.map{ meta, cns -> [meta, cns[2], []]})
+
+    // export to VCF for compatibility with other tools
+    CNVKIT_EXPORT(CNVKIT_CALL.out.cns)
+
```
Comment (FriederikeHanssen) [inline subworkflows/local/bam_variant_calling_cnvkit/main.nf:34]: we are missing the version calls for both

File: subworkflows/local/bam_variant_calling_cnvkit/main.nf [hunk 3]
```diff
@@ -21,12 +23,23 @@ workflow BAM_VARIANT_CALLING_CNVKIT {
...
     ch_genemetrics = CNVKIT_BATCH.out.cnr.join(CNVKIT_BATCH.out.cns).map{ meta, cnr, cns -> [meta, cnr, cns[2]]}
     CNVKIT_GENEMETRICS(ch_genemetrics)
 
     versions = versions.mix(CNVKIT_BATCH.out.versions)
     versions = versions.mix(CNVKIT_GENEMETRICS.out.versions)
```
Comment (FriederikeHanssen) [inline subworkflows/local/bam_variant_calling_cnvkit/main.nf:40]:
```suggestion
 versions = versions.mix( CNVKIT_CALL.out.versions)
 versions = versions.mix(  CNVKIT_EXPORT.out.versions)
```

File: workflows/sarek/main.nf [hunk 4]
```diff
@@ -149,10 +154,34 @@ workflow SAREK {
 
         // Figure out if input is bam or fastq
```
Comment (FriederikeHanssen) [inline workflows/sarek/main.nf:155]:
```suggestion
        // Figure out if input is bam, fastq, or spring
```

File: workflows/sarek/main.nf [hunk 5]
```diff
@@ -149,10 +154,34 @@ workflow SAREK {
...
+            true // write_one_fastq_gz
+        )
+        r2_fastq_gz_from_spring = SPRING_DECOMPRESS_TO_R2_FQ(input_sample_type.two_fastq_gz_spring.map{ meta, files ->
+            [meta, files[1] ]},
+            true // write_one_fastq_gz
+        )
+
+        two_fastq_gz_from_spring = r1_fastq_gz_from_spring.fastq.join(r2_fastq_gz_from_spring.fastq).map{ meta, fastq_1, fastq_2 -> [meta, [fastq_1, fastq_2]]}
+
+        two_fastq_gz_from_spring = two_fastq_gz_from_spring.map { meta, files -> addReadgroupToMeta(meta, files) }
```
Comment (FriederikeHanssen) [inline workflows/sarek/main.nf:183]: not entirely clear to me why this was not kept in the samplesheet subworkflow, but we can refactor this also later

**Review by FriederikeHanssen — APPROVED** [review]
Tentatively approvig. Waiting for the full size tests to pass

File: workflows/sarek/main.nf (hunk 5 above)
Comment (asp8200) [reply workflows/sarek/main.nf:183]: Yeah, I just remember we ran into problems with placing that stuff in the samplesheet subworkflow, and so we put it there.

File: workflows/sarek/main.nf (hunk 4 above)
Comment (asp8200) [reply workflows/sarek/main.nf:155]: I tried applying/commiting your suggestion, but GitHub complained.

**Comment by FriederikeHanssen** [comment]
@nf-core bot fix linting

**Comment by FriederikeHanssen** [comment]
@nf-core-bot fix linting

**Review by kenibrewer — APPROVED** [review]

**Comment by FriederikeHanssen** [comment]
tested the download locally with the latest tools release. @mashehu gave me permission to release with the failing test: https://nfcore.slack.com/archives/CE5LG7WMB/p1722964196047559?thread_ts=1722873465.605869&cid=CE5LG7WMB

**Review by asp8200 — APPROVED** [review]

**Pull request merged**